### PR TITLE
Adding NFTAccessCondition

### PR DESCRIPTION
**Pull request by aaitor**
## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: contracts/conditions/NFTs/NFTAccessCondition.sol [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    {
+        nftPermissions[_documentId].permission[_grantee] = true;
+    }
+
+   /**
+    * @notice renouncePermission is called only by DID owner or provider
+    * @param _grantee is the address of the granted user or the DID provider
+    * @param _documentId refers to the DID in which secret store will issue the decryption keys
+    */
+    function renouncePermission(
```
Comment (r-marques) [inline contracts/conditions/NFTs/NFTAccessCondition.sol:168]: We need to be sure that this is called when the nft is transfered, right?

File: contracts/conditions/NFTs/NFTAccessCondition.sol [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+ * @dev Implementation of the Access Condition specific for NFTs
+ *
+ *      NFT Access Condition is special condition used to give access 
+ *      to a specific NFT related to a DID.
+ */
+contract NFTAccessCondition is Condition {
+
+    bytes32 constant public CONDITION_TYPE = keccak256('NFTAccessCondition');
+
+    struct DocumentPermission {
```
Comment (r-marques) [inline contracts/conditions/NFTs/NFTAccessCondition.sol:24]: couldn't this be just a list with permitted addresses?

**Review by r-marques — APPROVED** [review]
lgtm

File: contracts/conditions/NFTs/NFTAccessCondition.sol (hunk 1 above)
Comment (aaitor) [reply contracts/conditions/NFTs/NFTAccessCondition.sol:168]: I think this function is not necessary, so I'm removing it

File: contracts/conditions/NFTs/NFTAccessCondition.sol (hunk 2 above)
Comment (aaitor) [reply contracts/conditions/NFTs/NFTAccessCondition.sol:24]: it's implemented in this way to replicate the existing AccessCondition

File: contracts/conditions/NFTs/NFTAccessCondition.sol (hunk 1 above)
Comment (r-marques) [reply contracts/conditions/NFTs/NFTAccessCondition.sol]: but a user should be able to access the nft after he transfers ownership of it, right?

Comment (aaitor) [reply contracts/conditions/NFTs/NFTAccessCondition.sol]: exactly, the first user will be able to access during the duration of the service agreement. It's difficult to rely on the first user renouncing voluntary to an access he has.

**Pull request merged**